mesonbuild/mparser.py

```diff
@@ -194,11 +193,10 @@ def lex(self, filename: str) -> T.Generator[Token, None, None]:
                     elif tid in {'string', 'fstring'}:
                         # Handle here and not on the regexp to give a better error message.
                         if match_text.find("\n") != -1:
-                            msg = ParseException("Newline character in a string detected, use ''' (three single quotes) "
-                                                 "for multiline strings instead.\n"
-                                                 "This will become a hard error in a future Meson release.",
-                                                 self.getline(line_start), lineno, col)
-                            mlog.warning(msg, location=BaseNode(lineno, col, filename))
+                            msg = ("Newline character in a string detected, use ''' (three single quotes) "
+                                   "for multiline strings instead.\n"
+                                   "This will become a hard error in a future Meson release.")
+                            mlog.warning(mlog.code_line(msg, self.getline(lineno), col))
```

Comment:
Alright, just tried testing this and it no longer even prints the correct line.

```suggestion
                            mlog.warning(mlog.code_line(msg, self.getline(line_start), col), location=BaseNode(lineno,col, filename))
```